File: test_cases/c/claude/02.c
#include <assert.h>
#include <stdbool.h>

bool isMatch(char* s, char* p);

int main(void) {
    assert(isMatch("aa", "a") == false);
    assert(isMatch("aa", "a*") == true);
    assert(isMatch("ab", ".*") == true);
    assert(isMatch("mississippi", "mis*is*p*.") == false);
    assert(isMatch("aab", "c*a*b") == true);
    return 0;
}

File: test_cases/python/claude/02.py
def run_tests(solution_module):
    f = solution_module.isMatch
    # zero repetitions via star
    assert f('b', 'a*b') is True
    # empty string matched by star
    assert f('', 'a*') is True
    # dot matches any char
    assert f('a', '.') is True
    # complex: star eliminates prefix
    assert f('aab', 'c*a*b') is True
    # fails without enough chars
    assert f('ab', 'a') is False
